frontend/pages/admin/IntegrationsPage/cards/ConditionalAccess/components/OktaConditionalAccessModal/OktaConditionalAccessModal.tsx

```diff
@@ -0,0 +1,396 @@
+import React, { useCallback, useContext, useEffect, useState } from "react";
+import { size } from "lodash";
+
+import { NotificationContext } from "context/notification";
+import { AppContext } from "context/app";
+import configAPI from "services/entities/config";
+import conditionalAccessAPI from "services/entities/conditional_access";
+import { IConfig } from "interfaces/config";
+import endpoints from "utilities/endpoints";
+
+// @ts-ignore
+import InputField from "components/forms/fields/InputField";
+import CustomLink from "components/CustomLink";
+import Modal from "components/Modal";
+import Button from "components/buttons/Button";
+import Icon from "components/Icon";
+import TooltipWrapper from "components/TooltipWrapper";
+import { IInputFieldParseTarget } from "interfaces/form_field";
+import { getErrorReason } from "interfaces/errors";
+import { LEARN_MORE_ABOUT_BASE_LINK } from "utilities/constants";
+import FileUploader from "components/FileUploader";
+
+const baseClass = "okta-conditional-access-modal";
+
+const OKTA_IDP_ID = "okta_idp_id";
+const OKTA_ACS_URL = "okta_assertion_consumer_service_url";
+const OKTA_AUDIENCE_URI = "okta_audience_uri";
+const OKTA_CERTIFICATE = "okta_certificate";
+
+interface IFormData {
+  [OKTA_IDP_ID]: string;
+  [OKTA_ACS_URL]: string;
+  [OKTA_AUDIENCE_URI]: string;
+  [OKTA_CERTIFICATE]: string;
+}
+
+interface IFormErrors {
+  [OKTA_IDP_ID]?: string | null;
+  [OKTA_ACS_URL]?: string | null;
+  [OKTA_AUDIENCE_URI]?: string | null;
+  [OKTA_CERTIFICATE]?: string | null;
+}
+
+const validate = (formData: IFormData) => {
+  const errs: IFormErrors = {};
+
+  // Max lengths from backend validation
+  const maxURLLength = 2048;
+  const maxCertLength = 8192;
+
+  // IdP ID validation - must be non-empty and not just whitespace
+  if (!formData[OKTA_IDP_ID] || !formData[OKTA_IDP_ID].trim()) {
+    errs[OKTA_IDP_ID] = "IdP ID must be present";
+  } else if (formData[OKTA_IDP_ID].length > maxURLLength) {
+    errs[OKTA_IDP_ID] = `IdP ID must be ${maxURLLength} characters or less`;
+  }
+
+  // Assertion Consumer Service URL validation
+  if (!formData[OKTA_ACS_URL] || !formData[OKTA_ACS_URL].trim()) {
+    errs[OKTA_ACS_URL] = "Assertion Consumer Service URL must be present";
+  } else if (formData[OKTA_ACS_URL].length > maxURLLength) {
+    errs[
+      OKTA_ACS_URL
+    ] = `Assertion Consumer Service URL must be ${maxURLLength} characters or less`;
+  } else {
+    // Validate URL format - must have http or https scheme and a host
```

Comment:
These conditions seem covered by our general URl validator, see `valid_url.ts`. Best to to use that
or extend its functionality as needed